xen/arch/arm/coproc/plat/coproc_xxx.c

```diff
@@ -336,18 +168,7 @@ static int coproc_xxx_dt_probe(struct platform_device *pdev)
 out_release_irqs:
     while ( i-- )
         release_irq(coproc_xxx->irqs[i], coproc_xxx);
-out_free_irqs:
-    xfree(coproc_xxx->irqs);
-out_iounmap_mmios:
-    for ( i = 0; i < num_mmios; ++i )
-    {
-        if ( !IS_ERR(coproc_xxx->mmios[i].base) )
-            iounmap(coproc_xxx->mmios[i].base);
-    }
-out_free_mmios:
-    xfree(coproc_xxx->mmios);
-    xfree(coproc_xxx);
-
+    coproc_release(coproc_xxx);
     return ret;
 }
 
```

Comment:
Some my comments from gx6xxx.c are applicable for coproc_xxx too. Please have a look